GenderPayGap.UnitTests/GenderPayGap.WebUI.Tests/Controllers/Account/ChangePasswordController/ChangePasswordControllerTests.cs

```diff
@@ -0,0 +1,187 @@
+            controller.AddMockUriHelperNew(new Uri("https://localhost:44371/mockURL").ToString());
+            
+            // Act
+            controller.ChangePasswordPost(new ChangePasswordNewViewModel()).Wait();
+            
+            // Assert
+            bool isExpectedPassword = await mockUserRepo.CheckPasswordAsync(user, "NewPassword1");
+            Assert.IsTrue(isExpectedPassword);
+        }
+        
+        [Test]
+        [Description("POST: Password is not updated when new password has fewer than 8 characters")]
+        public async Task POST_Password_Is_Not_Updated_When_New_Password_Has_Fewer_Than_8_Characters()
+        {
+            // Arrange
+            User user = new UserBuilder().WithPassword("password").Build();
+            
+            var requestFormValues = new Dictionary<string, StringValues>();
+            requestFormValues.Add("GovUk_Text_CurrentPassword", "password");
+            requestFormValues.Add("GovUk_Text_NewPassword", "Abc1");
+            requestFormValues.Add("GovUk_Text_ConfirmNewPassword", "Abc1");
+
+            var controller = new ControllerBuilder<ChangePasswordNewController>()
+                .WithUserId(user.UserId)
+                .WithRequestFormValues(requestFormValues)
+                .WithDatabaseObjects(user)
+                .Build();
+            
+            // Required to mock out the Url object when creating the verification URL
+            controller.AddMockUriHelperNew(new Uri("https://localhost:44371/mockURL").ToString());
+            
+            // Act
+            controller.ChangePasswordPost(new ChangePasswordNewViewModel()).Wait();
+            
+            // Assert
+            bool isExpectedPassword = await mockUserRepo.CheckPasswordAsync(user, "password");
+            Assert.IsTrue(isExpectedPassword);
+        }
+        
+        [Test]
+        [Description("POST: Password is not updated when new password is missing an uppercase letter")]
+        public async Task POST_Password_Is_Not_Updated_When_New_Password_Missing_Uppercase_Letter()
+        {
+            // Arrange
+            User user = new UserBuilder().WithPassword("password").Build();
+            
+            var requestFormValues = new Dictionary<string, StringValues>();
+            requestFormValues.Add("GovUk_Text_CurrentPassword", "password");
+            requestFormValues.Add("GovUk_Text_NewPassword", "abcdefg1");
+            requestFormValues.Add("GovUk_Text_ConfirmNewPassword", "abcdefg1");
+
+            var controller = new ControllerBuilder<ChangePasswordNewController>()
+                .WithUserId(user.UserId)
+                .WithRequestFormValues(requestFormValues)
+                .WithDatabaseObjects(user)
+                .Build();
+            
+            // Required to mock out the Url object when creating the verification URL
+            controller.AddMockUriHelperNew(new Uri("https://localhost:44371/mockURL").ToString());
+            

```

Comment:
It might be nice to add this to the controller builder
It looks like we don't care what the value of the URL is, so it would be great if we could avoid having `"https://localhost:44371/mockURL"` in this file (because it makes you think it's important to the test)